libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c

```diff
@@ -2000,8 +2003,9 @@ TEST( MQTT_Unit_API, GetConnectPacketSizeChecks )
 TEST( MQTT_Unit_API, SerializeConnectChecks )
 {
     IotMqttConnectInfo_t connectInfo;
+    IotMqttPublishInfo_t willInfo;
     size_t remainingLength = 0;
-    uint8_t buffer[ 20 ];
+    uint8_t buffer[ 70 ];
```

Comment:
We don't know how much space is needed until after a call to IotMqtt_GetConnectPacketSize.  To avoid having to use malloc, we allocate a buffer that should be large enough.

The test will fail if this buffer is not large enough:
```
TEST_ASSERT_GREATER_OR_EQUAL( packetSize, bufferSize );
```

Modifications that create a larger buffer would cause the test to fail, which is why I had to increase it.